format-avro/src/test/java/io/cdap/plugin/format/avro/input/AvroInputFormatProviderTest.java

```diff
@@ -70,4 +70,36 @@ public void testSchemaDetection() throws IOException {
     Schema schema = schemaDetector.detectSchema(avroFile.getPath(), formatContext, Collections.emptyMap());
     Assert.assertEquals(expectedSchema, schema);
   }
+
+  @Test
+  public void testPathFieldSchemaDetection() throws IOException {
+    // write avro file
+    Schema expectedSchema = Schema.recordOf("Avro",
+            Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+            Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+            Schema.Field.of("score", Schema.of(Schema.Type.DOUBLE)),
+            Schema.Field.of("test", Schema.of(Schema.Type.STRING)));
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+            .set("id", 0)
+            .set("name", "alice")
+            .set("score", 1.0d)
+            .set("test", "value")
```

Comment:
should not be writing this out as a data field, the path is something added, it is not in the data itself.

That being said, this is actually an edge case we need to handle. If the path field already exists in the data schema, we need to mark that as a validation error and tell the user to pick a different path field.

Reply:
Modified test to make sure exception arises if pathfield value is already in the schema